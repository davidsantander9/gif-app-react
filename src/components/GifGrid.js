import React, { useState, useEffect } from 'react'
import { getGifs } from '../helpers/getGifs';
import GifGridItem from './GifGridItem';

const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]); 

    useEffect( () => {
        getGifs(category)
        .then(setImages)
    }, [category]);
    // Ese arreglo es el component did mount

    return (
        <>
        <h3 className='animate__animated animate__flash animate__delay-1s'>{ category }</h3>
        <div className='card-grid'>
                {
                    images.map( (img) => {
                        return <GifGridItem 
                                key={img.id} 
                                { ...img }
                                />
                    })
                }
        </div>
        </>
    )
}

export default GifGrid;
