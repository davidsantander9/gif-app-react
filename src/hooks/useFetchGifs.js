import { useState, useEffect } from 'react'
import { getGifs } from '../helpers/getGifs';

const useFetchGifs = ( category ) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect( () => {
        getGifs(category)
        .then( imgs => {
            setState({
                data: imgs,
                loading: false
            })
        }

        ) 
    }, [category]); // Ese arreglo es el component did mount

    return state;
}

export default useFetchGifs;
