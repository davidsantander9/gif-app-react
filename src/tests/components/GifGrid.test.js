import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('<GifGrid/>  test', () => {

    const category = 'Batman';

    test('should correctly display the component <GifGrid/>', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true 
        });
        const wrapper = shallow(<GifGrid category={ category } />);
        expect( wrapper ).toMatchSnapshot();
    });

    test('should show when loads images', () => {
        const gifs = [{
            id: 'avas',
            url: 'https://localhost/cosa.png',
            title: 'A thing'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false 
        });

        const wrapper = shallow(<GifGrid category={ category } />);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    });
    
    
});