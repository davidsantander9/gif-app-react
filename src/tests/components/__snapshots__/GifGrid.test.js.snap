// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid/>  test should correctly display the component <GifGrid/> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flash animate__delay-1s"
  >
    Batman
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`<GifGrid/>  test should show when loads images 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flash animate__delay-1s"
  >
    Batman
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="avas"
      key="avas"
      title="A thing"
      url="https://localhost/cosa.png"
    />
  </div>
</Fragment>
`;
